__tests__/index.android.js

```diff
@@ -1,12 +1,34 @@
-import 'react-native';
+import {AppRegistry} from 'react-native';
 import React from 'react';
-import Index from '../index.android.js';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 
-it('renders correctly', () => {
-  const tree = renderer.create(
-    <Index />
-  );
+import Index from '../index.android.js';
+import UserGateway from '../App/gateway/user.js';
+
+describe('Android app', () => {
+    let navigation;
+
+    beforeEach(() => {
+        navigation = jest.fn();
+    });
+
+    it('renders correctly', () => {
+        const tree = renderer.create(
+            <Index navigation={navigation} />
+        );
+
+        expect(tree).toMatchSnapshot();
+    });
+
+    it('should log the user in', () => {
+        const spy = jest.spyOn(UserGateway, 'signIn');
+
+        renderer.create(
+            <Index navigation={navigation} />
+        );
+
+        expect(spy).toHaveBeenCalled()
```

Comment:
Just for future note, I prefer to test the effects, not the calls. I find spy testing to not be useful long term